@antoniospneto deu o mesmo erro do que no port para a [v15](https://github.com/OCA/l10n-brazil/pull/3194), se puder ver...

```
2024-07-28 13:17:37,519 361 ERROR odoo odoo.addons.l10n_br_fiscal.tests.test_fiscal_document_generic: FAIL: TestFiscalDocumentGeneric.test_nfe_same_state
Traceback (most recent call last):
  File "/__w/l10n-brazil/l10n-brazil/l10n_br_fiscal/tests/test_fiscal_document_generic.py", line 181, in test_nfe_same_state
    self.assertEqual(self.nfe_same_state.amount_price_gross, 3965)
AssertionError: 430.5 != 3965
```

--- a/l10n_br_fiscal/models/document_fiscal_line_mixin.py
+++ b/l10n_br_fiscal/models/document_fiscal_line_mixin.py
@@ -196,9 +196,12 @@ class FiscalDocumentLineMixin(models.AbstractModel):
     )
 
     price_gross = fields.Monetary(
+        string="Gross Product/Service Amount",
+        help=(
+            "Total value of products or services (quantity x unit price)"
+            "before any discounts."
+        ),
         compute="_compute_amounts",
-        string="Amount Gross",
-        help="Amount without discount.",
     )
 
     amount_untaxed = fields.Monetary(
@@ -223,7 +226,8 @@ class FiscalDocumentLineMixin(models.AbstractModel):
     )
 
     financial_total_gross = fields.Monetary(
-        string="Amount Financial Gross",
+        string="Financial Gross Amount",
+        help="Total amount before any discounts are applied.",
         compute="_compute_amounts",
     )
 

--- a/l10n_br_fiscal/models/document_fiscal_line_mixin_methods.py
+++ b/l10n_br_fiscal/models/document_fiscal_line_mixin_methods.py
@@ -131,19 +131,13 @@ class FiscalDocumentLineMixinMethods(models.AbstractModel):
     def _compute_amounts(self):
         for record in self:
             round_curr = record.currency_id or self.env.ref("base.BRL")
-            # Valor dos produtos
 
-            amount_subtotal = round_curr.round(record.price_unit * record.quantity)
+            # Total value of products or services
+            record.price_gross = round_curr.round(record.price_unit * record.quantity)
 
-            record.price_gross = (
-                amount_subtotal
-                + record.amount_tax_not_included
-                - record.amount_tax_withholding
-            )
+            record.amount_untaxed = record.price_gross - record.discount_value
 
-            record.amount_untaxed = amount_subtotal - record.discount_value
-
-            record.amount_fiscal = amount_subtotal - record.discount_value
+            record.amount_fiscal = record.price_gross - record.discount_value
 
             record.amount_tax = record.amount_tax_not_included
 

--- a/l10n_br_fiscal/tests/test_fiscal_document_generic.py
+++ b/l10n_br_fiscal/tests/test_fiscal_document_generic.py
@@ -158,6 +158,9 @@ class TestFiscalDocumentGeneric(TransactionCase):
                 " Básica to COFINS 3% de Venda de Contribuinte Dentro do Estado.",
             )
 
+            product_total = line.price_unit * line.quantity
+            self.assertEqual(line.price_gross, product_total)
+
         self.nfe_same_state.action_document_confirm()
 
         self.assertEqual(
@@ -173,6 +176,9 @@ class TestFiscalDocumentGeneric(TransactionCase):
             SITUACAO_EDOC_AUTORIZADA,
             "Document is not in Authorized state",
         )
+
+        # Total value of the products
+        self.assertEqual(self.nfe_same_state.amount_price_gross, 3965)
 
         result = self.nfe_same_state.action_document_cancel()
         self.assertTrue(result)

--- a/l10n_br_fiscal/tests/test_fiscal_document_generic.py
+++ b/l10n_br_fiscal/tests/test_fiscal_document_generic.py
@@ -49,7 +49,16 @@ class TestFiscalDocumentGeneric(TransactionCase):
         self.nfe_same_state._onchange_fiscal_operation_id()
 
         for line in self.nfe_same_state.fiscal_line_ids:
+            # Save the original price_unit value of the line as defined in
+            # the NFe demo data.
+            original_price_unit = line.price_unit
+
             line._onchange_product_id_fiscal()
+
+            # Restore the original price_unit value,
+            # as the product change might have altered it.
+            line.price_unit = original_price_unit
+
             line._onchange_commercial_quantity()
             line._onchange_ncm_id()
             line._onchange_fiscal_operation_id()
@@ -178,7 +187,7 @@ class TestFiscalDocumentGeneric(TransactionCase):
         )
 
         # Total value of the products
-        self.assertEqual(self.nfe_same_state.amount_price_gross, 3965)
+        self.assertEqual(self.nfe_same_state.amount_price_gross, 200)
 
         result = self.nfe_same_state.action_document_cancel()
         self.assertTrue(result)

